packages/tracing/src/transaction.ts

```diff
@@ -14,7 +14,7 @@ export class Transaction extends SpanClass implements TransactionInterface {
    */
   private readonly _hub: Hub = (getCurrentHub() as unknown) as Hub;
 
-  private readonly _trimEnd?: boolean;
+  private _trimEnd?: boolean;
```

Comment:
Since this is kind of an internal bit, is there a reason we'd want people to be able to muck with it? Same goes for `traceId`, `parentSpanId`, and the two timestamps. It seems like we could dump/update a subset of the properties rather than all of them, and then we wouldn't need this change. 

(To be fair, this same question applies to the original `beforeNavigate` - perhaps we should only pass a partial context there.)

Reply:
I agree that a user shouldn't mess with these but I think the reason they are exposed, and that we should expose them (but not document them) for downstream SDKs like React Native and Capacitor to use.

Reply:
I guess it depends on if there's a use case for `toContext` beyond user-generated functions like `beforeNavigate`. If wrapper SDKs use them themselves, then yeah, okay, but there's also no reason they can't modify the object directly, is there? I thought the only reason to do this was because we're trying to match `beforeNavigate`'s signature. No?

Reply:
That is true, the initial reason that I came to this was to mimic `beforeNavigate` and the RN SDK could not set `_trimEnd` as it was private.

Reply:
Then I'd vote for:
1) do whatever you need to to `_trimEnd` to make the RN SDK able to do its job
2) separately, have the methods above dump and read in only a subset of the data (whatever it makes sense to have a user touch in `beforeNavigate`). It occurs to me this also maybe could be used in what's passed to `tracesSampler`.

None of this is that critical, though, so your call.

Reply:
Hmm so I think the `updateWithContext` methods in this PR should be left as is, and the dumping should be done by `beforeNavigate` and `tracesSampler` separately